Pull request template (#2)
Libra-foundation/.github: PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md

## Prerequisites

> ***This part help you to don't forget anything before create pull request, If everything seem good you could delete it***

Reviewer: ```suggestion
***This part help you to don't forget anything before create pull request, If everything seem good you can delete it***
```

> Thanks for contribution :heart:. Before creating pull

Reviewer: ```suggestion
Thanks for your contribution :heart:. Before creating pull
```

> request you were able to check :

Reviewer: ```suggestion
requestplease make sure that:
```

- [ ] Your code follows the style guidelines of this project

> - [ ] You have performed a self-review of my code

Reviewer: ```suggestion
- [ ] You have performed a self-review of your code
```

> - [ ] You have commented my code, particularly in hard-to-understand areas

Reviewer: ```suggestion
- [ ] You have commented your code, particularly in hard-to-understand areas
```

- [ ] You have made corresponding changes to the documentation
- [ ] Your changes generate no new warnings

> - [ ] You have added tests that prove my fix is effective or that my feature works

Reviewer: ```suggestion
- [ ] You have added tests that prove your fix is effective or that your feature works
```

- [ ] New and existing unit tests pass locally with your changes
- [ ] Any dependent changes have been merged and published in downstream modules

> Your pull request title must contain type of change plus a short description. You have the choice between :

Reviewer: Title tags are more often in the for `[TAG]` where the tag is minimal. I do not think that's a great idea to restrict to a list. Adding some examples is a good way of ensuring common tags are respected in a gentle way.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

Example of title : `Bug fix : Pinneaples on pizza were removed`

Your pull request need to hold following parts.

## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List
any dependencies that are required for this change.

> ## How Has This Been Tested?

Reviewer: This section does not feel useful, we rely on unit tests and/or integration tests. The answer should always be "I created the relevant test cases".

> Disclaimer :exclamation: Issues are formatted using markdown, a useful summary of the features provided by GitHub can be found here.

Reviewer: Missing link.

## Prerequisites

***This part help you to don't forget anything before create pull request, If everything seem good you can delete it***

> Thanks for your contribution :heart:. Before creating pull

Reviewer: ```suggestion
Thanks for your contribution :heart:. Before creating a pull
```

request please make sure :

- [ ] Your code follows the style guidelines of this project
- [ ] You have performed a self-review of your code
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have made corresponding changes to the documentation
- [ ] Your changes generate no new warnings
- [ ] You have added tests that prove your fix is effective or that your feature works
- [ ] New and existing unit tests pass locally with your changes
- [ ] Any dependent changes have been merged and published in downstream modules

> Your pull request title must contain type of change plus a short description. Title type must be relevant and minimal (one or two words), this is a non-exhaustive list:

Reviewer: The tag must be relevant and minimal, the title can be longer than two words.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

> - Breaking change (fix or feature that would cause existing functionality to not work as expected)

Reviewer: I am more a fan of this type of tags :

```suggestion
- [BUG]: bug fix
- [FEATURE]: changes which adds functionality
- [BREAKING]: prepend this tag to any PR that breaks retro-compatibility.
```
I prefer them as they are shorter and clearly distinct from the title.
What do you prefer ?

***This part help you to don't forget anything before create pull request, If everything seem good you can delete it***

Thanks for your contribution :heart:. Before creating a pull request please make sure you checked all conditions

> of [checks part](#Checks), includes all following parts and title respect rule.

Reviewer: ```suggestion
of the [checks part](#Checks), including all following parts and the title rules as this makes our lives easier :heart:.
```

### Pull request title :

> Your pull request title must contain a tag type plus a short description. Tag must be relevant and minimal, this is a

Reviewer: ```suggestion
Your pull request title must contain a tag type plus a short description. Tags must be relevant and minimal, here is a
```

Example of title : `[BUG] Pinneaples on pizza were removed`

## 1) Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List
any dependencies that are required for this change.

## 2) Technical choice

Why you choose this method of implementation ? Why not another one ? 

## Checks

- [ ] Your code follows the style guidelines of this project
- [ ] You have performed a self-review of your code
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have made corresponding changes to the documentation
- [ ] Your changes generate no new warnings
- [ ] You have added tests that prove your fix is effective or that your feature works
- [ ] New and existing unit tests pass locally with your changes

> - [ ] Any dependent changes have been merged and published in downstream modules

Reviewer: I do not think this one is really needed. I mean, the down streamed modules should update in response to the approved changes of the repo, not the opposite, right ?
